source/fem-dem/cfd_dem_coupling.cc

```diff
@@ -894,6 +889,11 @@ CFDDEMSolver<dim>::dem_contact_build(unsigned int counter)
       this->pcout << "DEM contact search at dem step " << counter << std::endl;
       contact_build_number++;
 
+      // Particles displacement if passing through a periodic boundary
+      periodic_boundaries_object.execute_particles_displacement(
```

Comment:
As we discussed, I would add the if here, outside the function scope. I feel like it is cleaner to call a function only if it makes sense to call it instead of deciding if it is used within itself. Maybe doing "if bc is periodic" would make it more readable, but if you have "periodic_boundaries_cells_information.empty()" in this scope you can use it here to prevent looping through BCs. If empty() is the choice here, I would comment on it just to make sure you are testing if there are periodic BCs.

Reply:
Since the function is also used in DEM, I prefer to add the check for periodic boundary, also it's more straight forward.